packages/field-base/src/helper-controller.js

```diff
@@ -66,8 +66,9 @@ export class HelperController extends SlotController {
   setHelperText(helperText) {
     this.helperText = helperText;
 
-    if (!this.node || this.node === this.defaultNode) {
-      this.__applyDefaultHelper(helperText, this.defaultNode);
+    const helperNode = this.getSlotChild();
+    if (!helperNode || helperNode === this.defaultNode) {
```

Comment:
Check for `!helperNode` seems to be missing a test

Reply:
Good catch, added a test to cover this case (restoring the default helper).